src/components/landing/layout/footer/language-picker.tsx

```diff
@@ -1,62 +1,68 @@
 import moment from 'moment'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Form } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
 
+const languages = {
+  en: 'English',
+  'zh-CN': '简体中文',
+  'zh-TW': '繁體中文',
+  fr: 'Français',
+  de: 'Deutsch',
+  ja: '日本語',
+  es: 'Español',
+  ca: 'Català',
+  el: 'Ελληνικά',
+  pt: 'Português',
+  it: 'Italiano',
+  tr: 'Türkçe',
+  ru: 'Русский',
+  nl: 'Nederlands',
+  hr: 'Hrvatski',
+  pl: 'Polski',
+  uk: 'Українська',
+  hi: 'हिन्दी',
+  sv: 'Svenska',
+  eo: 'Esperanto',
+  da: 'Dansk',
+  ko: '한국어',
+  id: 'Bahasa Indonesia',
+  sr: 'Cрпски',
+  vi: 'Tiếng Việt',
+  ar: 'العربية',
+  cs: 'Česky',
+  sk: 'Slovensky'
+}
+
+const findLanguageCode = (wantedLanguage: string): string => {
+  const foundLanguage = Object.keys(languages).find((supportedLanguage) => {
+    return wantedLanguage === supportedLanguage || wantedLanguage.substr(0, 2) === supportedLanguage
+  })
+  return foundLanguage || ''
+}
+
 const LanguagePicker: React.FC = () => {
   const { i18n } = useTranslation()
 
-  const onChangeLang = async (event: React.ChangeEvent<HTMLSelectElement>) => {
+  const onChangeLang = useCallback(() => async (event: React.ChangeEvent<HTMLSelectElement>) => {
     const language = event.currentTarget.value
     moment.locale(language)
     await i18n.changeLanguage(language)
-  }
-
-  const shortenLanguageCode = (language: string): string => {
-    const languageWithoutArea = language.substr(0, 2)
-    if (languageWithoutArea === 'zh') {
-      return language
-    } else {
-      return languageWithoutArea
-    }
```

Comment:
brauchen wir den AreaCode nicht mehr?

Reply:
Mit dem neuen Code wird direkt der Testwert vom Browser verglichen einmal mit den Keys der Liste direkt und einmal der Substring ohne Area-Code mit den Keys der Liste.

Reply:
```javascript
return wantedLanguage === supportedLanguage || wantedLanguage.substr(0, 2) === supportedLanguage
```

Reply:
We don't need this specific case anymore. I made it a bit more general.

Reply:
guys, english please